dangerzone/document.py

```diff
@@ -72,6 +78,11 @@ def validate_output_filename(filename: str) -> None:
             # in unwriteable directory
             raise errors.UnwriteableOutputDirException()
 
+    def validate_default_archive_dir(self) -> None:
+        """Checks if archive dir can be created"""
+        if not os.access(self.default_archive_dir.parent, os.W_OK):
+            raise errors.UnwriteableArchiveDirException()
```

Comment:
There's something that worries me here. I tried to test this scenario out with the GUI version of Dangerzone. I created a directory, copy-pasted a document there, and then made the directory read-only.

After that, clicking on "Convert to Safe Document" did not do anything. In the terminal (since I started it via `dev_scripts/dangerzone`), I could see the `UnwriteableArchiveDirException` that is logged, but I'm afraid that the user won't see it. 

Triggered by that, I tried to see what would happen if we attempted to save a document in an output directory that is also not writable. In that case, we rely also on Python exceptions. What I saw is that the operation succeeded, but:

* the document was never written to the output directory (expected)
* no error was reported, either in the UI or the terminal logs (unexpected)

which is false, and the operation shouldn't start in the first place.

Reply:
Oh, right. We're missing the GUI detection for this. I will do that now.

Reply:
Done in 0ab5996

Reply:
Just took a look. Two things:

1. I see that the tooltip appears on hover everywhere on the window. Is this intentional?
2. Does this solve the problem of an unwriteable output directory for the safe files? Granted, this existed before, but I'd like to know.

Reply:
1.  Fixed in https://github.com/freedomofpress/dangerzone/pull/255/commits/fe0cd2771a278e331515a1d158707e338993d2f9
2. Good point. It doesn't and that will be problematic. Apparently it fails silently, saying it succeeded but didn't actually save the -safe.pdf file. It's probably a problem that already existed in previous release. We'd need to add logic to tell the user to change directory. Should be easy, should I do that? Or drop both?

Reply:
1. Nice, it works.
2. Hm, maybe I can give it a shot in a separate PR.

Reply:
Not sure if you did that, but I have already some working code. Did you end up working on it?

Reply:
Nope, I didn't. If you have some working code, then shoot, but I'd suggest in a different PR.